src/repository/pg.rs

```diff
@@ -75,6 +77,31 @@ impl IPaymentAddressRepository for PgPaymentAddressRepository {
 
         Ok(())
     }
+
+    async fn remove_payment_address(
+        &self,
+        domain: &str,
+        username: &str,
+        token: &str,
+    ) -> Result<()> {
+        use diesel::prelude::*;
+
+        let mut conn = self.pool.get()?;
+
+        let rows_deleted = diesel::delete(
+            payment_addresses::table
+                .filter(payment_addresses::domain.eq(domain))
+                .filter(payment_addresses::username.eq(username))
+                .filter(payment_addresses::authentication_token.eq(token)),
+        )
+        .execute(&mut conn)?;
```

Comment:
Is it idiomatic to do both the selection of entry to remove and the authentication check in the same SQL query? At least we can't distinguish no entry being present vs. incorrect auth this way. 

I'd probably build it as a three step process:
* Fetch entry
* Check auth
* Delete entry